src/Schema/Directives/BaseDirective.php

```diff
@@ -103,7 +103,12 @@ public function getMethodArgument(string $argumentName): \Closure
             throw new DirectiveException("Directive '{$this->name()}' must have an argument '{$argumentName}' with 'ClassName@methodName'");
         }
 
-        $className = $this->namespaceClassName($argumentParts[0]);
+        $className = $this->namespaceClassName($argumentParts[0], [
```

Comment:
Given this schema:

```graphql
type Query {
  foo: Int field(resolver: "Foo@foo")
}

type Mutation {
  foo: Int field(resolver: "Foo@foo")
}
```

Now, if you have `App\Http\GraphQL\Queries\Foo` and `App\Http\GraphQL\Mutations\Foo`, this will always match the latter.